### PR TITLE
LPS-104365 Track ResourceBundle to change modificationDate since JS can be generated using ResourceBundle keys.

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by vicnate5** [comment]
ci:test

**Comment by vicnate5** [comment]
@shuyangzhou Test results are good.

File: modules/apps/portal-language/portal-language-extender/src/main/java/com/liferay/portal/language/extender/internal/LanguageResourceServiceTrackerCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ */
+@Component(immediate = true, service = {})
+public class LanguageResourceServiceTrackerCustomizer
+	implements ServiceTrackerCustomizer<ResourceBundle, ResourceBundle> {
+
+	@Override
+	public ResourceBundle addingService(
+		ServiceReference<ResourceBundle> serviceReference) {
+
+		ResourceBundle resourceBundle = _bundleContext.getService(
```
Comment (shuyangzhou) [inline modules/apps/portal-language/portal-language-extender/src/main/java/com/liferay/portal/language/extender/internal/LanguageResourceServiceTrackerCustomizer.java]: There is no matching ungetService() call.

File: modules/apps/portal-template/portal-template-soy-impl/src/main/java/com/liferay/portal/template/soy/internal/SoyManager.java [hunk 2]
```diff
@@ -160,4 +182,44 @@ protected void setSoyTemplateBundleResourceParser(
...
+			if (Validator.isNotNull(languageId)) {
+				locale = LocaleUtil.fromLanguageId(languageId, true);
+			}
+			else {
+				locale = new Locale(StringPool.BLANK);
+			}
+
+			_soyTofuCacheHandler.removeIfAny(locale);
+
+			return _bundleContext.getService(serviceReference);
```
Comment (shuyangzhou) [inline modules/apps/portal-template/portal-template-soy-impl/src/main/java/com/liferay/portal/template/soy/internal/SoyManager.java]: There is no matching ungetService() call.

File: modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/JavaScriptPortalWebResources.java [hunk 3]
```diff
@@ -65,6 +87,37 @@ protected void setServletContext(ServletContext servletContext) {
...
+	private class LanguageResourceServiceTrackerCustomizer
+		implements ServiceTrackerCustomizer<ResourceBundle, ResourceBundle> {
+
+		@Override
+		public ResourceBundle addingService(
+			ServiceReference<ResourceBundle> serviceReference) {
+
+			_resourceBundleLastModified = System.currentTimeMillis();
+
+			return _bundleContext.getService(serviceReference);
```
Comment (shuyangzhou) [inline modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/JavaScriptPortalWebResources.java]: There is no matching ungetService() call.

**Comment by shuyangzhou** [comment]
Do you really need to use a SreviceTracker? You don't ever use the collected stuff. I think a ServiceListener is good enough for this.

And timestamp is very weak here. All you need is a way to check whether a change has happened, change it to use ids generated by an AtomicLong, that way it won't ever suffer a timestamp conflict.

Please resend.

**Pull request closed without merging**